api/v0/proto/common.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package celestia.node.v1.common;
+
+option go_package = "github.com/celestiaorg/celestia-node/api/v1/common";
+
+// Common status codes used across all services
+enum StatusCode {
```

Comment:
Would it make sense to follow the precedent of HTTP status codes? [2xx for success, 3xx redirect, 4xx client error, 5xx server error](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status). I think it would just be more familiar and intuitive, unless there's gRPC reasons to follow a different convention.